### ISSUE TITLE
differences with async-profiler

### ISSUE THREAD

**Issue by jiacai2050** [riemann/riemann-jvm-profiler#9]
[async-profiler](https://github.com/jvm-profiling-tools/async-profiler) says it avoids safepoint problem in its README and a low-level profile for jvm-applications.

I'm curious if riemann-jvm-profiler also avoid safepoint, and what's more or less compared with async-profile, Thanks in advance.

**Comment by aphyr** [comment]
Nope, those APIs didn't exist at the time I wrote this. You're welcome to port async-profiler's methods to Riemann-JVM-profiler, though.On Apr 9, 2018 03:17, Jiacai Liu <notifications@github.com> wrote:async-profiler says it avoids safepoint problem in its README and a low-level profile for jvm-applications.
I'm curious if riemann-jvm-profiler also avoid safepoint, and what's more or less compared with async-profile, Thanks in advance.

**Comment by goldshtn** [comment]
As far as I can tell from the source code, it uses ThreadMXBean to get stack traces, which is subject to the safepoint bias.

**Comment by jiacai2050** [comment]
Thank your guys, I will try to port async-profile's method to Riemann-JVM-profiler.
